R/check_tbl_spl_non_compound_tid.R

```diff
@@ -0,0 +1,102 @@
+#' Check model output data tbl samples contain single unique combination of
+#' non-compound task ID values across all samples
+#' @param tbl a tibble/data.frame of the contents of the file being validated. Column types must **all be character**.
+#' @inherit check_tbl_colnames params
+#' @inherit check_tbl_colnames return
+#' @details Output of the check includes an `errors` element, a list of items,
+#' one for each modeling task containing samples failing validation,
+#' with the following structure:
+#' - `mt_id`: Index identifying the config modeling task the samples are associated with.
+#' - `output_type_ids`: The output type IDs of samples that do not match the most prevalent
+#' non-compound task ID value combination across all
+#' samples in the modeling task.
+#' - `prevalent`: The most prevalent non-compound task ID value combination
+#' across all samples in the modeling task.
+#' See [hubverse documentation on samples](https://hubverse.io/en/latest/user-guide/sample-output-type.html)
+#' for more details.
+#' @export
+check_tbl_spl_non_compound_tid <- function(tbl, round_id, file_path, hub_path) {
+  config_tasks <- hubUtils::read_config(hub_path, "tasks")
+
+  if (isFALSE(has_spls_tbl(tbl)) || isFALSE(hubUtils::is_v3_config(config_tasks))) {
+    return(skip_v3_spl_check(file_path))
+  }
+
+  hash_tbl <- spl_hash_tbl(tbl, round_id, config_tasks)
+
+  n_tbl <- dplyr::summarise(
+    hash_tbl,
+    n = dplyr::n_distinct(.data$hash_non_comp_tid),
+    mt_id = unique(.data$mt_id)
+  ) %>%
+    dplyr::filter(.data$n > 1L)
+
+  check <- nrow(n_tbl) == 0L
+
+  if (check) {
+    details <- NULL
+    errors <- NULL
+  } else {
+    errors <- non_comptid_mismatch_errors(
+      mt_ids = n_tbl$mt_id, hash_tbl, tbl,
+      config_tasks, round_id
+    )
+    output_type_ids <- purrr::map(errors, ~ .x$output_type_ids) %>% # nolint: object_usage_linter
+      unlist(use.names = FALSE)
+
+    details <- cli::format_inline(
+      "Sample{?s} {.val {output_type_ids}} d{?oes/o} not match most prevalent ",
+      "non compound task ID combination for {?its/their} modeling task. ",
+      "See {.var errors} attribute for details."
```

Comment:
As the compound ID error message was updated, should we also update that one so that they "match"? Maybe something like: 
```
Sample{?s} {.val {output_type_ids}} d{?oes/o} not match ",
      "the expected non compound task ID combinations.",
      "See {.var errors} attribute for details."
 ```

Reply:
Here it is comparing non compound task id combinations (which can include optional values - i.e. in this test there is no baseline to check against but a number of potential combinations across all rows of a sample which all samples need to share) to the most prevalent combination and identifying ones that don't match. So this is actually an accurate description of what the check is doing.

Reply:
I thought about it because the error message changed for the other function (which makes sense), but also the documentation of this function changed to `"Check model output data tbl samples contain single unique combination of non-compound task ID values across all samples"` so I thought it was making sense to update the error message to "match" non only with the other function style but also the title of the documentation here. I see now that `expected` was not the best description. Also if I am the only one having trouble with `prevalent` here, feel free to ignore. I was confused by it before, but after more explanation, `prevalent` makes sense to me now, so my only hesitation was that I was not sure if it would make sense for the users.

Reply:
I don't feel error messages need to repeat the same information throughout verbatim. The samples do need to be consistent but it is important to state how we determine that consistency and how we identify which are the deviations from that consistency. Because of that it is important to mention that we compare to the most prevalent combination. Having said that, because it might just be the word "prevalent" that is the problem, I've changed it to "frequent" throughout which hopefully is more widely used and understood?

Reply:
Perfect, thank you very much.